File: src/test/java/by/miaskor/java/creational/factory_method/factory/BoatFactoryTest.java
package by.miaskor.java.creational.factory_method.factory;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import by.miaskor.java.creational.factory_method.domain.Boat;
import by.miaskor.java.creational.factory_method.domain.Transport;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class BoatFactoryTest {

  private final TransportFactory transportFactory = mock(BoatFactory.class);

  @Test
  void make() {
    Transport expected = new Boat();
    when(transportFactory.make())
        .thenReturn(expected);
    Transport actual = transportFactory.make();
    Assertions.assertEquals(expected, actual);
  }
}

File: src/test/kotlin/by/miaskor/kotlin/creational/factory_method/factory/BoatFactoryTest.kt
package by.miaskor.kotlin.creational.factory_method.factory

import by.miaskor.kotlin.creational.factory_method.domain.Boat
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

internal class BoatFactoryTest {

  private val transportFactory = mock(BoatFactory::class.java)

  @Test
  fun make() {
    val expected = Boat()
    `when`(transportFactory.make())
      .thenReturn(expected)
    val actual = transportFactory.make()
    Assertions.assertEquals(expected, actual)
  }
}
